tutorials/relay/from_onnx.py

```diff
@@ -0,0 +1,103 @@
+"""
+Compile ONNX Models
+===================
+**Author**: `Joshua Z. Zhang <https://zhreshold.github.io/>`_
+
+This article is an introductory tutorial to deploy ONNX models with Relay.
+
+For us to begin with, onnx module is required to be installed.
+
+A quick solution is to install protobuf compiler, and
+
+.. code-block:: bash
+
+    pip install onnx --user
+
+or please refer to offical site.
+https://github.com/onnx/onnx
+"""
+import onnx
+import numpy as np
+import tvm
+import tvm.relay as relay
+
+def download(url, path, overwrite=False):
+    import os
+    if os.path.isfile(path) and not overwrite:
+        print('File {} existed, skip.'.format(path))
+        return
+    print('Downloading from url {} to {}'.format(url, path))
+    try:
+        import urllib.request
+        urllib.request.urlretrieve(url, path)
+    except:
+        import urllib
+        urllib.urlretrieve(url, path)
+
+######################################################################
+# Load pretrained ONNX model
+# ---------------------------------------------
+# The example super resolution model used here is exactly the same model in onnx tutorial
+# http://pytorch.org/tutorials/advanced/super_resolution_with_caffe2.html
+# we skip the pytorch model construction part, and download the saved onnx model
+model_url = ''.join(['https://gist.github.com/zhreshold/',
+                     'bcda4716699ac97ea44f791c24310193/raw/',
+                     '93672b029103648953c4e5ad3ac3aadf346a4cdc/',
+                     'super_resolution_0.2.onnx'])
+download(model_url, 'super_resolution.onnx', False)
+# now you have super_resolution.onnx on disk
+onnx_model = onnx.load('super_resolution.onnx')
+
+######################################################################
+# Load a test image
+# ---------------------------------------------
+# A single cat dominates the examples!
+from PIL import Image
+img_url = 'https://github.com/dmlc/mxnet.js/blob/master/data/cat.png?raw=true'
+download(img_url, 'cat.png')
+img = Image.open('cat.png').resize((224, 224))
+img_ycbcr = img.convert("YCbCr")  # convert to YCbCr
+img_y, img_cb, img_cr = img_ycbcr.split()
+x = np.array(img_y)[np.newaxis, np.newaxis, :, :]
+
+######################################################################
+# Compile the model with relay
+# ---------------------------------------------
+target = 'llvm'
+
+input_name = '1'
+shape_dict = {input_name: x.shape}
+sym, params = relay.frontend.from_onnx(onnx_model, shape_dict)
+
+with relay.build_config(opt_level=1):
+    graph, lib, params = relay.build(sym, target, params=params)
+
+######################################################################
```

Comment:
Can we maybe just show how to use the `create_executor` interface in the tutorial as well? its much simpler then building a graph runtime by hand

Reply:
@jroesch  Can you please advice how to elegantly evaluate executor created with an unordered dict of weights?

```python

exec = relay.build_module.create_executor('graph', mod, tvm.cpu(0), 'llvm')
# params is a dict of {'name', tvm.ndarray} 
tvm_output = exec.evaluate(mode)(x, *params.values()).asnumpy()   # wrong order
```
How do I feed the executor with correct weights without changing the params dict to odict?

Reply:
@zhreshold I realize the current API does not actually support keyword argument style for parameters. 

I'm going to update it with a PR right now, it should work like this:

```
exec = relay.build_module.create_executor('graph', mod, tvm.cpu(0), 'llvm')
tvm_output = exec.evaluate(mode)(x, **params).asnumpy()
```

Does that seem elegant enough?

Reply:
Yes, that would perfectly solve my current problem

Reply:
Will update once the PR get merged.

Reply:
PR is merged.